src/query-builder/InsertQueryBuilder.ts

```diff
@@ -347,6 +347,18 @@ export class InsertQueryBuilder<Entity> extends QueryBuilder<Entity> {
             query += ` RETURNING ${returningExpression}`;
         }
 
+
+        // Inserting a specific value for an auto-increment primary key in mssql requires enabling IDENTITY_INSERT
+        // IDENTITY_INSERT can only be enabled for tables where there is an IDENTITY column and only if there is a value to be inserted (i.e. supplying DEFAULT is prohibited if IDENTITY_INSERT is enabled)
+        if (this.connection.driver instanceof SqlServerDriver
+            && this.expressionMap.mainAlias!.hasMetadata
+            && this.expressionMap.mainAlias!.metadata.columns
+                .filter((column) => this.expressionMap.insertColumns.length > 0 ? this.expressionMap.insertColumns.indexOf(column.propertyPath) !== -1 : column.isInsert)
+                .some((column) => this.isOverridingAutoIncrementBehavior(column))
+        ) {
+            query = `SET IDENTITY_INSERT ${tableName} ON; ${query}; SET IDENTITY_INSERT ${tableName} OFF`;
```

Comment:
Also, Instead of doing this as part of building the query itself, how would it look as part of the `execute` function?

That way it's not muddying up the SQL we've generated - and we already do something similar to that for `@OutputTable`.  That way it's more for the session than the query itself we're generating.

Reply:
I thought about that. However, i think its more appropriate to have it here. Otherwise, the query generation method will generate a query which cannot be run, unless you execute it with the execute function. In other words, this method would build an invalid query, without additional context provided by execute. There is also a lot of driver dependent code in this method anyways.

The OutputTable handling in the execute method on the other hand has nothing to do with the insert query, but the way we retrieve results. In other words, without the OutputTable handling, the query would still work, it just could not retrieve the results correctly (which is a concern with regard to the return value of the execute function).